<!DOCTYPE html>
<html>
    
    <head>
        <meta charset='utf-8' />
        <meta http-equiv="X-UA-Compatible" content="chrome=1" />
        <meta name="description" content="Ensembl API workshop : Ensembl core API, Whole-genome alignments, Ensembl compara API (genomic alignments)" />
        
        <link rel="stylesheet" type="text/css" media="screen" href="../stylesheets/stylesheet.css">
            
            <title id=doctitle>Whole Genome Alignments</title>
            </head>
    
    <body>
        <script type="text/javascript" src="http://www.ebi.ac.uk/inc/js/jquery.js"></script>
        <script type="text/javascript">
            $(document).ready(function() {
                              $("#header_wrap").load("header.html");
                              title = document.title;
                              $('#doctitle').load("title.txt", '', function(data) {
                                                  if ($(this).text() != title) {
                                                  document.title = $(this).text() + " - " + title
                                                  }
                                                  });
                              
                              });
            </script>
        
        <!-- HEADER -->
        <div id="header_wrap" class="outer">
            <header class="inner">
                <a id="forkme_banner" href="https://github.com/jherrero/EnsemblAPI_workshop/tree/2013-12.Madrid">View on GitHub</a>
                
                <h1 id="project_title">This Workshop</h1>
                
                <section id="downloads">
                    <a class="zip_download_link" href="https://github.com/jherrero/EnsemblAPI_workshop/zipball/master">Download this project as a .zip file</a>
                    <a class="tar_download_link" href="https://github.com/jherrero/EnsemblAPI_workshop/tarball/master">Download this project as a tar.gz file</a>
                </section>
            </header>
        </div>
        
        <!-- MAIN CONTENT -->
        <div id="main_content_wrap" class="outer">
            <section id="main_content" class="inner">
                
                <p><a href="index.html">Main page</a> &gt; <strong>Whole Genome Alignments</strong></p>
                

<h2>Whole Genome Alignments</h2>
<p>In this exercise, we will use MultiZ to align 5 Escherichia species and Pecan to align the BRCA2 locus in human, mouse, rat, cow, oposssum and chicken. Feel free to use another locus or a different set of species, although you would have to prepare the input files yourself.</p>
<p>One of the most annoying problems to run a whole genome aligner is to get the
information in the right format. Whole genome aligner will require two basic
pieces of information: the sequences to align and a species tree that relates the
sequences. Different softwares rely on different criteria to link the species tree
with the sequences. For instance, Pecan rely on the order of the sequences files, i.e.
the leaves of the tree must be in the same order as the input files. MultiZ, on the other
hand, relies on the name of the files.</p>


<h2>MultiZ for 5 Escherichia species</h2>
<p>This exercises proposes to align 5 Escherichia species using MultiZ.</p>
<p>MultiZ is a software that combines several pairwise of multiple alignments into one single alignment using a reference species.</p>
<h3>Here is the work done to prepare this exercise:</h3>
<ol>
<li>Select the set of species to be aligned. In this case I selected 5 Escherichia species from Ensembl Genomes (http://www.ensemblgenomes.org)</li>
<li>Get a species tree (branch length are not important at this stage). You may have to trim the tree if it contains additional species. I used Dendroscope for this.</li>
<li>Get the tree in the right format. With a text editor, I have modified the tree such as it does not contain comas, colons, distances nor the final semicolon</li>
<li>Download the (repeat-masked) sequences for the species of interest. I have downloaded them from ftp://ftp.ensemblgenomes.org/pub/bacteria/release-11/fasta/escherichia_shigella_collection/:
<ul>
<li>ftp://ftp.ensemblgenomes.org/pub/bacteria/release-11/fasta/escherichia_shigella_collection/e_coli_k12/dna/E_coli_k12.EB1_e_coli_k12.dna.toplevel.fa.gz</li>
<li>ftp://ftp.ensemblgenomes.org/pub/bacteria/release-11/fasta/escherichia_shigella_collection/e_coli_o17_k52_h18/dna/E_coli_o17_k52_h18.EB1_e_coli_o17_k52_h18.dna_rm.chromosome.Chromosome.fa.gz</li>
<!--
<li>ftp://ftp.ensemblgenomes.org/pub/bacteria/release-11/fasta/escherichia_shigella_collection/e_coli_o6/dna/E_coli_o6.EB1_e_coli_o6.dna_rm.chromosome.Chromosome.fa.gz</li>
-->
<li>ftp://ftp.ensemblgenomes.org/pub/bacteria/release-11/fasta/escherichia_shigella_collection/e_coli_o8/dna/E_coli_o8.EB1_e_coli_o8.dna_rm.chromosome.Chromosome.fa.gz</li>
<li>ftp://ftp.ensemblgenomes.org/pub/bacteria/release-11/fasta/escherichia_shigella_collection/e_coli_se15/dna/E_coli_se15.EB2_e_coli_se15.dna_rm.chromosome.Chromosome.fa.gz</li>
</ul>
</li>
<li>Change the name of the files to match the species names in the tree</li>
<!--
<li>Modify the E_coli_o6 to mask all non-canonical nucleotides</li>
-->
</ol>


<h3>Resources</h3>
<ul>
<li>MultiZ/TBA and blastz download site: <a href="http://www.bx.psu.edu/miller_lab/">http://www.bx.psu.edu/miller_lab/</a></li>
<li>Input files:
 <ul>
 <li>E_coli_o17_k52_h18 fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_o17_k52_h18">E_coli_o17_k52_h18</a>] </li>
 <li>E_coli_o8 fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_o8">E_coli_o8</a>] </li>
<!--
 <li>E_coli_o6 fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_o6">E_coli_o6</a>] </li>
-->
 <li>E_coli_se15 fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_se15">E_coli_se15</a>] </li>
 <li>E_coli_k12 fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_k12">E_coli_k12</a>] </li>
 </ul></li>
</ul>

<h3>Exercise:</h3>
<ol>
<!--
<li>Install blastz (used by MultiZ)
<pre>mkdir -p ~/Downloads
cd ~/Downloads
wget http://www.bx.psu.edu/miller_lab/dist/blastz-2004-12-27.deprecated.tar.gz
mkdir -p ~/bin
cd ~/bin
tar -zxvf ~/Downloads/blastz-2004-12-27.deprecated.tar.gz
cd blastz-source/
make
</pre>
</li>
-->

<li>We should install blastz, but it is deprecated. The substitute is called lastz. Let's install this:
<pre>mkdir -p ~/Downloads
cd ~/Downloads
wget http://www.bx.psu.edu/miller_lab/dist/lastz-1.02.00.tar.gz
mkdir -p ~/bin
cd ~/bin
tar -zxvf ~/Downloads/lastz-1.02.00.tar.gz
# We will ignore the warnings
sed -e 's/-Werror//' -i src/Makefile
cd lastz-distrib-1.02.00/
make
make install
make test

# Create a link for lastz called blastz
ln -s ~/lastz-distrib/bin/lastz ~/bin/blastz
</pre>
</li>

<li>Install multiz and tba:
<pre>cd ~/Downloads
wget http://www.bx.psu.edu/miller_lab/dist/multiz-tba.012109.tar.gz
cd ~/bin
tar -zxvf ~/Downloads/multiz-tba.012109.tar.gz
cd multiz-tba.012109/
# We will ignore the warnings
sed -e 's/-Werror//' -i Makefile
make
</pre>
</li>

<li>Now, modify your path to run MulitZ and blastz:
<pre>
export PATH=$PATH:~/bin/:~/bin/multiz-tba.012109/
</pre>
</li>

<li>Create a new directory and download the sequences. For instance:
<pre>mkdir -p ~/whole_genome_alignments
cd ~/whole_genome_alignments
mkdir -p multiz
cd multiz
wget <a id="E_coli_o17_k52_h18_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_o17_k52_h18</a>
wget <a id="E_coli_o8_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_o8</a>
<!--
 wget <a id="E_coli_o6_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_o6</a>
-->
wget <a id="E_coli_se15_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_se15</a>
wget <a id="E_coli_k12_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/E_coli_k12</a>
</pre>
</li>

<li>See what will be run:
<!--
<pre>
all_bz - F=E_coli_k12 D=0 "(E_coli_o17_k52_h18 (E_coli_o8 ((E_coli_o6 E_coli_se15) E_coli_k12)))"
</pre>
-->
<pre>
all_bz - F=E_coli_k12 D=0 "(E_coli_o17_k52_h18 (E_coli_o8 (E_coli_se15 E_coli_k12)))"
</pre>
<p>These creates all the blastz (pairwise alignment) commands required for MultiZ. It also creates all the self-alignments that are not required. We will ignore those. The last commands filter the result to remove overlapping alignments.</p>
<p>Let's try to get the same list of commands, but without the self-alignments:</p>
<pre>
all_bz - F=E_coli_k12 D=0 \
    "(E_coli_o17_k52_h18 (E_coli_o8 (E_coli_se15 E_coli_k12)))" \
    | grep -v "T=2"
</pre>
<!--
<pre>
all_bz - F=E_coli_k12 D=0 \
    "(E_coli_o17_k52_h18 (E_coli_o8 ((E_coli_o6 E_coli_se15) E_coli_k12)))" \
    | grep -v "T=2"
</pre>
-->

</li>

<li>Create a bash file to run this:
<!--
<pre>all_bz - F=E_coli_k12 D=0 \
    "(E_coli_o17_k52_h18 (E_coli_o8 ((E_coli_o6 E_coli_se15) E_coli_k12)))" \
    | grep -v "T=2" > run.all_bz
</pre>
-->
<pre>all_bz - F=E_coli_k12 D=0 \
    "(E_coli_o17_k52_h18 (E_coli_o8 (E_coli_se15 E_coli_k12)))" \
    | grep -v "T=2" > run.all_bz
</pre>
</li>

<li>Make this file executable.
<pre>chmod +x run.all_bz
</pre>
</li>

<li>Run it!
<pre>
./run.all_bz
</pre>
</li>

<li>The next step is fairly complex. Essentially, we have to run multiz for pairs of species/alignments in the order given by the species tree. Luckily, we have a program that makes everything much easier. It is called roast. See what it does:
<pre>
roast - E=E_coli_k12 \
    "(E_coli_o17_k52_h18 (E_coli_o8 (E_coli_se15 E_coli_k12)))" \
    Escherichia_shigella.multiz.maf
</pre>
<!--
<pre>
roast - E=E_coli_k12 \
    "(E_coli_o17_k52_h18 (E_coli_o8 ((E_coli_o6 E_coli_se15) E_coli_k12)))" \
    Escherichia_shigella.multiz.maf
</pre>
-->
</li>

<li>Create a new bash file for roast:
<pre>
roast - E=E_coli_k12 \
    "(E_coli_o17_k52_h18 (E_coli_o8 (E_coli_se15 E_coli_k12)))" \
    Escherichia_shigella.multiz.maf > run.roast
</pre>
<!--
<pre>
roast - E=E_coli_k12 \
    "(E_coli_o17_k52_h18 (E_coli_o8 ((E_coli_o6 E_coli_se15) E_coli_k12)))" \
    Escherichia_shigella.multiz.maf > run.roast
</pre>
-->
</li>

<li>Make this file executable.
<pre>chmod +x run.roast
</pre>
</li>

<li>Run it!
<pre>
./run.roast
</pre>
</li>
</ol>

<h3>Output</h3>
<p>Multiz and TBA use the MAF format [<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format5">description</a>].
This format supports several multiple alignments in the same file.</p>
<p>Final multiple alignment: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/MultiZ/Escherichia_shigella.multiz.maf">Escherichia_shigella.multiz.maf</a>]</p>




<h2>Pecan</h2>
<h3>Running the software</h3>
<p>Pecan is written in Java and requires Exonerate (a BLAST-like alignment software). The basic command line is:</p>
<pre>java -cp &lt;PECAN.JAR&gt; bp.pecan.Pecan -E &lt;TREE_STRING&gt; -F &lt;SORTED_FASTA_FILES&gt; -G &lt;OUTPUT_FILE&gt;</pre>
<p>Pecan wants a binary tree in Newick format. You can have distances, but they will be ignored. Pecan only
cares about the topology of the tree. Here is the tree for these species:</p>
<pre>((((HUMAN,(MOUSE,RAT)),COW),OPOSSUM),CHICKEN);</pre>
<p>Build the command line. Remember to quote the tree and provide the sequence files in the right order! You may have to specify the location to exonerate (/soft/exonerate-2.2.0-x86_64/bin/)</p>

<h3>Resources</h3>
<ul>
<li>Exonerate main web site: <a href="http://www.ebi.ac.uk/~guy/exonerate/">http://www.ebi.ac.uk/~guy/exonerate/</a></li>
<li>Pecan link: <a href="http://www.ebi.ac.uk/~jherrero/downloads/pecan_v0.8.jar">http://www.ebi.ac.uk/~jherrero/downloads/pecan_v0.8.jar</a></li>
<li>Pecan general README file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/README_FIRST.txt">README_FIRST.txt</a>]</li>
<li>Pecan README file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/README_PECAN.txt">README_PECAN.txt</a>]</li>
<li>Help for pecan command line: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/pecan.txt">pecan.txt</a>]</li>
<li>Input files for Pecan (UNIX text file format):
 <ul>
 <li>Human fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/human.fa">human.fa</a>] </li>
 <li>Mouse fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/mouse.fa">mouse.fa</a>] </li>
 <li>Rat fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/rat.fa">rat.fa</a>] </li>
 <li>Cow fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/cow.fa">cow.fa</a>] </li>
 <li>Opossum fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/opossum.fa">opossum.fa</a>] </li>
 <li>Chicken fasta file: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/chicken.fa">chicken.fa</a>] </li>
 </ul></li>
</ul>

<h3>Exercise</h3>
<ol>
<li>Install Pecan
<pre>mkdir -p ~/bin
cd ~/bin
wget http://www.ebi.ac.uk/~jherrero/downloads/pecan_v0.8.jar
</pre>
</li>

<li>Create a new directory for this exercise:
<pre>mkdir -p ~/whole_genome_alignments
cd ~/whole_genome_alignments
mkdir -p pecan
cd pecan
wget <a id="human_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/human.fa</a>
wget <a id="mouse_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/mouse.fa</a>
wget <a id="rat_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/rat.fa</a>
wget <a id="cow_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/cow.fa</a>
wget <a id="opossum_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/opossum.fa</a>
wget <a id="chicken_file_url">https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/chicken.fa</a>
</pre>
</li>
<li>Run Pecan:
<pre>
java -cp ~/bin/pecan_v0.8.jar bp.pecan.Pecan \
    -E "((((HUMAN,(MOUSE,RAT)),COW),OPOSSUM),CHICKEN);" \
    -F human.fa mouse.fa rat.fa cow.fa opossum.fa chicken.fa \
    -G pecan.mfa
</pre>
</li>

</ol>

<h3>Output</h3>
<p>Pecan's output is a fasta alignment file.</p>
<p>Final multiple alignment: [<a href="https://raw.github.com/jherrero/EnsemblAPI_workshop/2013-12.Madrid/Exercises/Pecan/pecan.mfa">pecan.mfa</a>] (NB: different runs can give different results)</p>

<h3>Troubleshooting</h3>
<ul>
<li>Make sure that the order of the sequence files match the order of the leavs in the tree.</li>
</ul>

<hr>
<hr>
<p>Next exercise: <a href="Conservation.html">Genomic Sequence Conservation</a></p>



            </section>
        </div>
        
        <!-- FOOTER  -->
        <div id="footer_wrap" class="outer">
            <footer class="inner">
                <p class="copyright">Ensembl API workshop maintained by <a href="https://github.com/jherrero">jherrero</a></p>
                <p>Published with <a href="http://pages.github.com">GitHub Pages</a></p>
            </footer>
        </div>
        
        
        
    </body>
</html>
