<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" 
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!-- Generated by perltidy on Tue Jan 31 11:23:53 2012 -->
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>genomic_align_block1b.pl</title>
<style type="text/css">
<!--
/* default style sheet generated by perltidy */
body {background: #FFFFFF; color: #000000}
pre { color: #000000; 
      background: #FFFFFF;
      font-family: courier;
    } 

.c  { color: #228B22;} /* comment */
.cm { color: #000000;} /* comma */
.co { color: #000000;} /* colon */
.h  { color: #CD5555; font-weight:bold;} /* here-doc-target */
.hh { color: #CD5555; font-style:italic;} /* here-doc-text */
.i  { color: #00688B;} /* identifier */
.j  { color: #CD5555; font-weight:bold;} /* label */
.k  { color: #8B008B; font-weight:bold;} /* keyword */
.m  { color: #FF0000; font-weight:bold;} /* subroutine */
.n  { color: #B452CD;} /* numeric */
.p  { color: #000000;} /* paren */
.pd { color: #228B22; font-style:italic;} /* pod-text */
.pu { color: #000000;} /* punctuation */
.q  { color: #CD5555;} /* quote */
.s  { color: #000000;} /* structure */
.sc { color: #000000;} /* semicolon */
.v  { color: #B452CD;} /* v-string */
.w  { color: #000000;} /* bareword */
-->
</style>
</head>
<body>
<a name="-top-"></a>
<h1>genomic_align_block1b.pl</h1>
<hr />
<!-- contents of filename: genomic_align_block1b.pl -->
<pre>
<span class="c">#</span>
<span class="c"># This script prints the LASTZ_NET alignments for pig chromosome 15 with cow</span>
<span class="c"># using pig coordinates 89151597 and 89157190)</span>
<span class="c">#</span>
<span class="k">use</span> <span class="w">strict</span><span class="sc">;</span>
<span class="k">use</span> <span class="w">warnings</span><span class="sc">;</span>
<span class="k">use</span> <span class="w">Bio::AlignIO</span><span class="sc">;</span>

<span class="k">use</span> <span class="w">Bio::EnsEMBL::Registry</span><span class="sc">;</span>

<span class="c">#Auto-configure the registry</span>
<span class="w">Bio::EnsEMBL::Registry</span><span class="w">-&gt;load_registry_from_db</span><span class="s">(</span>
	-<span class="w">host</span><span class="cm">=&gt;</span><span class="q">&quot;ensembldb.ensembl.org&quot;</span><span class="cm">,</span>
	-<span class="w">user</span><span class="cm">=&gt;</span><span class="q">&quot;anonymous&quot;</span><span class="s">)</span><span class="sc">;</span>


<span class="c"># Get the Compara Adaptor for MethodLinkSpeciesSets</span>
<span class="k">my</span> <span class="i">$method_link_species_set_adaptor</span> =
    <span class="w">Bio::EnsEMBL::Registry</span><span class="w">-&gt;get_adaptor</span><span class="s">(</span>
      <span class="q">&quot;Multi&quot;</span><span class="cm">,</span> <span class="q">&quot;compara&quot;</span><span class="cm">,</span> <span class="q">&quot;MethodLinkSpeciesSet&quot;</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># Get the MethodLinkSpecieSet for pig-cow lastz-net alignments</span>
<span class="k">my</span> <span class="i">$methodLinkSpeciesSet</span> = <span class="i">$method_link_species_set_adaptor</span>-&gt;
	<span class="w">fetch_by_method_link_type_registry_aliases</span><span class="s">(</span><span class="q">&quot;LASTZ_NET&quot;</span><span class="cm">,</span> <span class="s">[</span><span class="q">&quot;pig&quot;</span><span class="cm">,</span> <span class="q">&quot;cow&quot;</span><span class="s">]</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># Define the start and end positions for the alignment</span>
<span class="k">my</span> <span class="s">(</span><span class="i">$pig_start</span><span class="cm">,</span> <span class="i">$pig_end</span><span class="s">)</span> = <span class="s">(</span><span class="n">89151597</span><span class="cm">,</span> <span class="n">89157190</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># Get the pig *core* Adaptor for Slices</span>
<span class="k">my</span> <span class="i">$pig_slice_adaptor</span> =
    <span class="w">Bio::EnsEMBL::Registry</span><span class="w">-&gt;get_adaptor</span><span class="s">(</span>
      <span class="q">&quot;sus_scrofa&quot;</span><span class="cm">,</span> <span class="q">&quot;core&quot;</span><span class="cm">,</span> <span class="q">&quot;Slice&quot;</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># Get the slice corresponding to the region of interest</span>
<span class="k">my</span> <span class="i">$pig_slice</span> = <span class="i">$pig_slice_adaptor</span><span class="i">-&gt;fetch_by_region</span><span class="s">(</span>
    <span class="q">&quot;chromosome&quot;</span><span class="cm">,</span> <span class="q">&quot;15&quot;</span><span class="cm">,</span> <span class="i">$pig_start</span><span class="cm">,</span> <span class="i">$pig_end</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># Get the Compara Adaptor for GenomicAlignBlocks</span>
<span class="k">my</span> <span class="i">$genomic_align_block_adaptor</span> =
    <span class="w">Bio::EnsEMBL::Registry</span><span class="w">-&gt;get_adaptor</span><span class="s">(</span>
      <span class="q">&quot;Multi&quot;</span><span class="cm">,</span> <span class="q">&quot;compara&quot;</span><span class="cm">,</span> <span class="q">&quot;GenomicAlignBlock&quot;</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># The fetch_all_by_MethodLinkSpeciesSet_Slice() returns a ref.</span>
<span class="c"># to an array of GenomicAlingBlock objects (pig is the reference species) </span>
<span class="k">my</span> <span class="i">$all_genomic_align_blocks</span> = <span class="i">$genomic_align_block_adaptor</span>-&gt;
    <span class="w">fetch_all_by_MethodLinkSpeciesSet_Slice</span><span class="s">(</span>
        <span class="i">$methodLinkSpeciesSet</span><span class="cm">,</span> <span class="i">$pig_slice</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># set up an AlignIO to format SimpleAlign output</span>
<span class="k">my</span> <span class="i">$alignIO</span> = <span class="w">Bio::AlignIO</span><span class="w">-&gt;newFh</span><span class="s">(</span>-<span class="w">interleaved</span> <span class="cm">=&gt;</span> <span class="n">0</span><span class="cm">,</span>
                                  -<span class="w">fh</span> <span class="cm">=&gt;</span> \<span class="i">*STDOUT</span><span class="cm">,</span>
                                  -<span class="w">format</span> <span class="cm">=&gt;</span> <span class="q">&#39;clustalw&#39;</span><span class="cm">,</span>
                                  -<span class="w">idlength</span> <span class="cm">=&gt;</span> <span class="n">20</span><span class="s">)</span><span class="sc">;</span>

<span class="c"># print the restricted alignments</span>
<span class="k">foreach</span> <span class="k">my</span> <span class="i">$genomic_align_block</span> <span class="s">(</span> <span class="i">@</span>{ <span class="i">$all_genomic_align_blocks</span> } <span class="s">)</span> <span class="s">{</span>
	<span class="k">my</span> <span class="i">$restricted_gab</span> = <span class="i">$genomic_align_block</span><span class="i">-&gt;restrict_between_reference_positions</span><span class="s">(</span><span class="i">$pig_start</span><span class="cm">,</span> <span class="i">$pig_end</span><span class="s">)</span><span class="sc">;</span>
	<span class="k">print</span> <span class="i">$alignIO</span> <span class="i">$restricted_gab</span><span class="i">-&gt;get_SimpleAlign</span><span class="sc">;</span>
<span class="s">}</span>

</pre>
</body>
</html>
